src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -1278,21 +1278,18 @@ private TResult WriteToOutput<TResult>(ODataPayloadKind payloadKind, Func<ODataO
         /// <param name="payloadKind">The payload kind to write.</param>
         /// <param name="writeAsyncAction">The write operation to invoke on the output.</param>
         /// <returns>Task which represents the pending write operation.</returns>
-        private Task WriteToOutputAsync(ODataPayloadKind payloadKind, Func<ODataOutputContext, Task> writeAsyncAction)
+        private async Task WriteToOutputAsync(ODataPayloadKind payloadKind, Func<ODataOutputContext, Task> writeAsyncAction)

```

Comment:
Could you explain the change here. Seems like in this method you did the opposite of eliding async :)

Reply:
This is a private method and the local function within it will _always_ be executed. Local function would have added value if there was a branch of logic that didn't execute the enclosed set of statements. I therefore got rid of the local function and made the private function async.